### ISSUE TITLE
java.util.ConcurrentModificationException on broadcast

### ISSUE THREAD

**Issue by papirosko** [TooTallNate/Java-WebSocket#961]
Hi. Broadcasting uses connections without synchronization:

```
	/**
	 * Send a text to all connected endpoints
	 * @param text the text to send to the endpoints
	 */
	public void broadcast(String text) {
		broadcast( text, connections );
	}

```

even trying to override this method with:
```
  override void broadcast(String: text) = {
    broadcast( text, getConnections() );
  }
```

will lead to

```
java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1445)
	at java.util.HashMap$KeyIterator.next(HashMap.java:1469)
	at java.util.AbstractCollection.toArray(AbstractCollection.java:141)
	at java.util.ArrayList.<init>(ArrayList.java:178)
	at org.java_websocket.server.WebSocketServer.getConnections(WebSocketServer.java:282)
```

**Comment by papirosko** [comment]
as a workaround, use `new CopyOnWriteArraySet()` in websocket constructor.

**Comment by izamata01** [comment]
can you guys help me get my mobile back in my control. someone is
redirecting me. My phone company have no idea. Customer service is all i
get.Its almost like someones cloned my phone. cloned my phone. Plz help me.
im not a developer im just reaching out for help

**Comment by marci4** [comment]
Hello,
Could you try the latest snapshot?
Should be fixed with #922.

Best regards,
Marcel

**Comment by marci4** [comment]
Closing this issue as duplicate!

Best regards,
Marcel
